### PR TITLE
docs: Add and update test specifications for `AccountUpdateTransaction`

### LINKED ISSUE #207: Update `AccountUpdateTransaction` documentation
Update all the docs for `AccountUpdateTransaction` to the specifications and conventions we have established. When approved and merged, the tests should be updated to match what is described in the docs, as well as naming conventions, etc.

### PR DESCRIPTION
**Pull request by rwalworth**
This PR updates the test specification for `AccountUpdateTransaction`. Tests have been added to test the bounds and ranges of `AccountUpdateTransaction` properties, as well as match the style and guidelines of our test specification files.

This PR also adds a `privateKeys` return value for `generateKey` for `keyList` and `thresholdKey` types. This is useful for signing a transaction with a `keyList` or `thresholdKey` type, as the private keys are needed and were previously unobtainable.

**Note**: This is waiting on changes to be merged in #209.

**Related issue(s)**:

Fixes #207

### CODE REVIEW

File: test-specifications/crypto-service/accountUpdateTransaction.md
Comment (thenswan) [inline test-specifications/crypto-service/accountUpdateTransaction.md:45]: Link to a json object is not valid.

File: test-specifications/crypto-service/accountUpdateTransaction.md
Comment (thenswan) [inline test-specifications/crypto-service/accountUpdateTransaction.md:105]: Can we test minimum and maximum amount of keys in a KeyList?

File: test-specifications/crypto-service/accountUpdateTransaction.md
Comment (thenswan) [inline test-specifications/crypto-service/accountUpdateTransaction.md:153]: Should be 'plus one second' (also there are similar places like this one).

File: test-specifications/crypto-service/accountUpdateTransaction.md
Comment (thenswan) [inline test-specifications/crypto-service/accountUpdateTransaction.md:67]: I think should be 'The account update fails' here (also there are similar instances across this file, where 'account creation' should be changed to 'account update').

**Review by thenswan — CHANGES_REQUESTED** [review]
GJ, just a couple of edits 🙌🏻

File: test-specifications/crypto-service/accountUpdateTransaction.md
Comment (rwalworth) [reply test-specifications/crypto-service/accountUpdateTransaction.md:45]: Yeah this links to a markdown that's not merged yet (#208). You approved it but I requested a re-review because I added a lot of changes after you approved. Once that's merged this should be fine.

File: test-specifications/crypto-service/accountUpdateTransaction.md
Comment (rwalworth) [reply test-specifications/crypto-service/accountUpdateTransaction.md]: I guess I missed some edits from when I copy-pasted from `accountCreateTransaction.md`.. fixed

File: test-specifications/crypto-service/accountUpdateTransaction.md
Comment (rwalworth) [reply test-specifications/crypto-service/accountUpdateTransaction.md]: I guess I missed some edits from when I copy-pasted from `accountCreateTransaction.md`.. fixed

File: test-specifications/crypto-service/accountUpdateTransaction.md
Comment (rwalworth) [reply test-specifications/crypto-service/accountUpdateTransaction.md:45]: Made a note in the PR description about this.

File: test-specifications/crypto-service/accountUpdateTransaction.md
Comment (thenswan) [reply test-specifications/crypto-service/accountUpdateTransaction.md:105]: Not sure tbh, this needs checking

File: test-specifications/crypto-service/accountUpdateTransaction.md
Comment (thenswan) [reply test-specifications/crypto-service/accountUpdateTransaction.md:45]: Oh, got it, okay

File: test-specifications/crypto-service/accountUpdateTransaction.md
Comment (thenswan) [reply test-specifications/crypto-service/accountUpdateTransaction.md:105]: In that case, I think that we can drop testing with the maximum number of keys in a KeyList. Tyvm for checking that out 👍🏻

**Review by SimiHunjan — APPROVED** [review]

**Comment by rwalworth** [comment]
@thenswan were there any changes you still requested or could I get an approval to merge this?

**Review by thenswan — APPROVED** [review]
LGTM

**Pull request merged**
